### ISSUE TITLE
[BUG] dp compress fail

### ISSUE THREAD

**Issue by neojie** [deepmodeling/deepmd-kit#1299]
version: deepmd-kit 2.0.3, cpu version
description: dp compress -i pv.pb -o comp.pb -t input_v2_compat.json 
error:
```
2021-11-19 15:00:00.761448: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
WARNING:tensorflow:From /u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
DEEPMD INFO    stage 0: compute the min_nbor_dist
2021-11-19 15:00:14.806285: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-11-19 15:00:14.809741: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /u/home/j/jd848/project-lstixrud/plumed/lib:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/compiler/lib/intel64_lin:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/mpi/intel64/libfabric/lib:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/mpi/intel64/lib/release:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/mpi/intel64/lib:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/ipp/lib/intel64:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/mkl/lib/intel64_lin:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/tbb/lib/intel64/gcc4.8:/u/local/compilers/intel/2020.4/debugger_2020/python/intel64/lib:/u/local/compilers/intel/2020.4/debugger_2020/libipt/intel64/lib:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/daal/lib/intel64_lin:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/daal/../tbb/lib/intel64_lin/gcc4.4:/u/local/compilers/intel/2020.4/compilers_and_libraries_2020.4.304/linux/daal/../tbb/lib/intel64_lin/gcc4.8
2021-11-19 15:00:14.809792: W tensorflow/stream_executor/cuda/cuda_driver.cc:326] failed call to cuInit: UNKNOWN ERROR (303)
2021-11-19 15:00:14.809854: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (g32): /proc/driver/nvidia/version does not exist
2021-11-19 15:00:14.809932: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 12. Tune using inter_op_parallelism_threads for best performance.
2021-11-19 15:00:14.814628: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 2666785000 Hz
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
# warning: ghost idx out of upper bound (ignored if warned for more than 10 times) 
DEEPMD INFO    training data with min nbor dist: 0.2582362501611707
DEEPMD INFO    training data with max nbor size: [42, 40, 123, 320]
DEEPMD INFO    


DEEPMD INFO    stage 1: compress the model
DEEPMD INFO     _____               _____   __  __  _____           _     _  _   
DEEPMD INFO    |  __ \             |  __ \ |  \/  ||  __ \         | |   (_)| |  
DEEPMD INFO    | |  | |  ___   ___ | |__) || \  / || |  | | ______ | | __ _ | |_ 
DEEPMD INFO    | |  | | / _ \ / _ \|  ___/ | |\/| || |  | ||______|| |/ /| || __|
DEEPMD INFO    | |__| ||  __/|  __/| |     | |  | || |__| |        |   < | || |_ 
DEEPMD INFO    |_____/  \___| \___||_|     |_|  |_||_____/         |_|\_\|_| \__|
DEEPMD INFO    Please read and cite:
DEEPMD INFO    Wang, Zhang, Han and E, Comput.Phys.Comm. 228, 178-184 (2018)
DEEPMD INFO    installed to:         /tmp/pip-req-build-i991qczn/_skbuild/linux-x86_64-3.9/cmake-install
DEEPMD INFO    source :              v2.0.3
DEEPMD INFO    source brach:         HEAD
DEEPMD INFO    source commit:        159e45d2
DEEPMD INFO    source commit at:     2021-10-15 10:31:09 +0800
DEEPMD INFO    build float prec:     double
DEEPMD INFO    build with tf inc:    /u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/tensorflow/include
DEEPMD INFO    build with tf lib:    
DEEPMD INFO    ---Summary of the training---------------------------------------
DEEPMD INFO    running on:           g32
DEEPMD INFO    computing device:     cpu:0
DEEPMD INFO    CUDA_VISIBLE_DEVICES: unset
DEEPMD INFO    Count of visible GPU: 0
DEEPMD INFO    num_intra_threads:    0
DEEPMD INFO    num_inter_threads:    0
DEEPMD INFO    -----------------------------------------------------------------
DEEPMD INFO    training with 1 frame parameter(s)
Traceback (most recent call last):
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/bin/dp", line 10, in <module>
    sys.exit(main())
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/deepmd/entrypoints/main.py", line 447, in main
    compress(**dict_args)
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/deepmd/entrypoints/compress.py", line 119, in compress
    train(
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 103, in train
    _do_work(jdata, run_opt, is_compress)
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 159, in _do_work
    model.build(train_data, stop_batch)
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/deepmd/train/trainer.py", line 285, in build
    self.descrpt.enable_compression(self.model_param['compress']["min_nbor_dist"], self.model_param['compress']['model_file'], self.model_param['compress']['table_config'][0], self.model_param['compress']['table_config'][1], self.model_param['compress']['table_config'][2], self.model_param['compress']['table_config'][3])
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/deepmd/descriptor/se_a.py", line 330, in enable_compression
    self.table = DPTabulate(
  File "/u/home/j/jd848/project-lstixrud/dp2/dp2gpuconda/lib/python3.9/site-packages/deepmd/utils/tabulate.py", line 106, in __init__
    for item in self.matrix["layer_" + str(self.layer_size)]:
KeyError: 'layer_0'
```
Another related issue is that I get very large (~300 M) output model file when using 2.0.3. If I use a different version, say v2.0.0.b1,  the frozen model is only around 10 M.

**Comment by denghuilu** [comment]
Could you please provide the original pb model for debugging？

**Comment by neojie** [comment]
Sure. The link below directs to the file. It is too large and I have to share it via google drive.
[https://drive.google.com/file/d/1KOUx-m-8BQftcs-lQXa02xSzmiQtHkAp/view?usp=sharing](url)
After unzip the file, you will see 
```
bash-3.2$ ls
dat v2
bash-3.2$ cd v2/
bash-3.2$ ls
checkpoint                     input_v2_compat.json           model.ckpt.data-00000-of-00001 model.ckpt.meta                out.json
compress.json                  lcurve.out                     model.ckpt.index               out.533262                     pv.pb
```
`dat` folder contains all the data while `v2` folder contains the training results. Within `v2`, `pv.pb` is the frozen model using `deepmd v2.0.0.b1` GPU version by `dp freeze -o pv.pb`. If you try freezing the model using `deepmd v2.0.1 cpu`, or `deepmd v2.0.2 cpu`, or `deepmd v2.0.3 gpu and cpu`, you will get a frozen model of over 140 M, compared with only ~10 M of pv.pb. I do not understand why the size difference is so large.
Anyway, I also tried to compress model  using `deepmd v2.0.1 cpu`, or `deepmd v2.0.2 cpu`, or `deepmd v2.0.3 gpu and cpu`, I got the error as shown in the first thread.

**Comment by denghuilu** [comment]
When I used the `compress.json` script to compress the `pv.pb` model within the v2 folder, I also confirmed the following error:
```
root v2 $ dp compress -i pv.pb -o pv-compress.pb --training-script compress.json 
WARNING:tensorflow:From /root/denghui/tensorflow_venv/lib/python3.7/site-packages/tensorflow/python/compat/v2_compat.py:101: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
WARNING:root:To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
DEEPMD INFO    stage 0: compute the min_nbor_dist
DEEPMD INFO    training data with min nbor dist: 0.620452881731422
DEEPMD INFO    training data with max nbor size: [33, 34, 111, 216]
/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/utils/compat.py:347: UserWarning: The argument training->numb_test has been deprecated since v2.0.0. Use training->validation_data->batch_size instead.
  "The argument training->numb_test has been deprecated since v2.0.0. "
DEEPMD INFO    


DEEPMD INFO    stage 1: compress the model
DEEPMD WARNING Switch to serial execution due to lack of horovod module.
DEEPMD INFO     _____               _____   __  __  _____           _     _  _   
DEEPMD INFO    |  __ \             |  __ \ |  \/  ||  __ \         | |   (_)| |  
DEEPMD INFO    | |  | |  ___   ___ | |__) || \  / || |  | | ______ | | __ _ | |_ 
DEEPMD INFO    | |  | | / _ \ / _ \|  ___/ | |\/| || |  | ||______|| |/ /| || __|
DEEPMD INFO    | |__| ||  __/|  __/| |     | |  | || |__| |        |   < | || |_ 
DEEPMD INFO    |_____/  \___| \___||_|     |_|  |_||_____/         |_|\_\|_| \__|
DEEPMD INFO    Please read and cite:
DEEPMD INFO    Wang, Zhang, Han and E, Comput.Phys.Comm. 228, 178-184 (2018)
DEEPMD INFO    installed to:         /root/denghui/deepmd-kit/_skbuild/linux-x86_64-3.7/cmake-install
DEEPMD INFO    source :              v2.0.2-31-g9c517eb-dirty
DEEPMD INFO    source brach:         model-compression-check
DEEPMD INFO    source commit:        9c517eb
DEEPMD INFO    source commit at:     2021-11-22 10:36:17 +0800
DEEPMD INFO    build float prec:     double
DEEPMD INFO    build with tf inc:    /root/denghui/tensorflow_venv/lib/python3.7/site-packages/tensorflow/include
DEEPMD INFO    build with tf lib:    
DEEPMD INFO    ---Summary of the training---------------------------------------
DEEPMD INFO    running on:           VM-0-4-centos
DEEPMD INFO    computing device:     gpu:0
DEEPMD INFO    CUDA_VISIBLE_DEVICES: unset
DEEPMD INFO    Count of visible GPU: 1
DEEPMD INFO    num_intra_threads:    0
DEEPMD INFO    num_inter_threads:    0
DEEPMD INFO    -----------------------------------------------------------------
DEEPMD INFO    training with 1 frame parameter(s)
DEEPMD INFO    training data with lower boundary: -0.9983819693372643
DEEPMD INFO    training data with upper boundary: 11.081454650233423
DEEPMD INFO    built lr
Traceback (most recent call last):
  File "/root/denghui/tensorflow_venv/bin/dp", line 33, in <module>
    sys.exit(load_entry_point('deepmd-kit==2.0.3.dev31+g9c517eb.d20211122', 'console_scripts', 'dp')())
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/entrypoints/main.py", line 447, in main
    compress(**dict_args)
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/entrypoints/compress.py", line 127, in compress
    is_compress=True,
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/entrypoints/train.py", line 103, in train
    _do_work(jdata, run_opt, is_compress)
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/entrypoints/train.py", line 159, in _do_work
    model.build(train_data, stop_batch)
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/train/trainer.py", line 294, in build
    self._build_network(data)
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/train/trainer.py", line 326, in _build_network
    reuse = False)
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/model/ener.py", line 192, in build
    suffix = suffix)
  File "/root/denghui/tensorflow_venv/lib/python3.7/site-packages/deepmd_kit-2.0.3.dev31+g9c517eb.d20211122-py3.7-linux-x86_64.egg/deepmd/fit/ener.py", line 363, in build
    raise RuntimeError('No data stat result. one should do data statisitic, before build')
RuntimeError: No data stat result. one should do data statisitic, before build
```
When numb_fparam is not zero, an error occurs when compressing the model.

**Comment by njzjz** [comment]
Fixed in #1306.
